bin/cyhy-domain

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+"""Manage domain assignments to owners.
+
+Usage:
+  cyhy-domain [options] (add | remove) OWNER [DOMAINS ...]
+  cyhy-domain [options] check [DOMAINS ...]
+  cyhy-domain [options] list OWNER
+  cyhy-domain [options] list-all
+  cyhy-domain [options] move OWNER NEW_OWNER [DOMAINS ...]
+  cyhy-domain (-h | --help)
+  cyhy-domain --version
+
+Options:
+  -c FILE --config-file=FILE     Configuration file to use.
+  -f FILENAME --file=FILENAME    Read domains from a file.
+  -h --help                      Show this screen.
+  -s SECTION --section=SECTION   Configuration section to use.
+  --version                      Show version.
+
+Notes:
+  Domains can be read from standard input if no domains are provided on the
+  command line and the --file option is omitted.
+"""
+
+# Standard Python Libraries
+import sys
+
+# Third-Party Libraries
+from docopt import docopt
+from validators import domain as valid_domain
+
+# cisagov Libraries
+from cyhy.core.common import *
+from cyhy.db import database
+from cyhy.util import util
+
+def parse_domains(domains):
+    parsed_domains = set()
+    invalid_domains = set()
+    for d in domains:
+        d = d.split("#", 1)[0]  # remove comments
+        d = d.strip()
+        if not d:
+            # skip blank lines
+            continue
+        if not valid_domain(d):
+            invalid_domains.add(d)
+        else:
+            parsed_domains.add(d)
+    
+    if invalid_domains:
+        print "ERROR - invalid domain(s):"
+        for d in invalid_domains:
+            print(d)
+        sys.exit(-1)
+    return parsed_domains
+
+
+def read_file(filename):
+    f = open(filename, "r")
+    return parse_domains(f.readlines())
+
+
+def do_list(db, owner):
+    request = db.RequestDoc.find_one({"_id": owner})
+    for h in request.get("hostnames", []):
+        print(h)
+
+
+def do_list_all(db):
+    requests = db.RequestDoc.find({"hostnames": {"$exists": True}}).sort("_id")
+    for r in requests:
+        print "# %s (%s): %d" % (
+            r["agency"]["name"],
+            r["_id"],
+            len(r["hostnames"])
+        )
```

Comment:
I'm confused as to why this script uses Python2-style `print` (`print` is not a function) whereas `cyhy-domainsync` uses Python3-style `print` (`print` is a function).  Both files have the same shebang so they should be running the same version of Python.

Reply:
OK, but the non-function syntax doesn't work in Python 3 and the function syntax doesn't work in Python 2.  Both scripts have the same shebang, so they are presumably running in the same Python interpreter.  It seems that one of them must fail just due to the `print` syntax.